Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilderSettings.cpp

```diff
@@ -7,17 +7,50 @@
  */
 
 #include <SdfAssetBuilder/SdfAssetBuilderSettings.h>
+#include <SdfAssetBuilder/SdfAssetBuilder.h>
 
 #include <AzCore/Serialization/Json/JsonUtils.h>
 #include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
 
 namespace ROS2
 {
-        namespace
+    namespace
+    {
+        struct SDFSettingsRootKeyType
         {
-            constexpr const char* SdfAssetBuilderSupportedFileExtensionsRegistryKey = "/O3DE/ROS2/SdfAssetBuilder/SupportedFileTypeExtensions";
-            constexpr const char* SdfAssetBuilderUseArticulationsRegistryKey = "/O3DE/ROS2/SdfAssetBuilder/UseArticulations";
-        }
+            using StringType = AZStd::fixed_string<256>;
+
+            constexpr StringType operator()(AZStd::string_view name) const
```

Comment:
Neat! I wanted to do this in a clean compile-time way but didn't know how to, this is pretty cool. :)

Reply:
Yeah, I originally came up with this kind of function object structure in the [EventLogger](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Metrics/IEventLogger.h#L30-L38) code